src/convertToObject.js

```diff
@@ -3,16 +3,43 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesString = sourceString.split(';');
+  const styles = stylesString.filter(style => style.trim())
+    .map(style => style.trim())
+    .map(style => style.split(':'))
+    .map(style => style.join(''))
+    .map(style => style.split(' '))
+    .map(style => style.filter(style1 => style1 !== ''));
+
+  const obj = {};
+
+  styles.forEach((style) => {
+    const name = style[0];
+    let value = style[1];
+
+    if (style[2]) {
```

Comment:
Instead of checking for specific indices, use a more general approach to concatenate the values.